ocs_ci/utility/aws.py

```diff
@@ -725,6 +744,153 @@ def verify_stack_deleted(stack_name):
+
+    def get_s3_bucket_object_url(self, bucket_name, object_key):
+        """
+        Get s3 bucket object url
+
+        Args:
+            bucket_name (str): Name of a valid s3 bucket
+            object_key (str): Name of the key for s3 object
+
+        Returns:
+            s3_url (str): An s3 url
+
+        """
+        s3_url = os.path.join(
+            f'https://{bucket_name}.s3.{self._region_name}.amazonws.com',
+            f'{object_key}'
+        )
+        return s3_url
+
+    def get_stack_instance_id(self, stack_name, logical_id):
+        """
+        Get the instance id associated with the cloudformation stack
+
+        Args:
+            stack_name (str): Name of the cloudformation stack
+            logical_id (str):  LogicalResourceId of the resource
+                ex: "Worker0"
+
+        Returns:
+            instance_id (str): Id of the instance
+
+        """
+        resource = self.cf_client(
+            StackName=stack_name, LogicalResourceId=logical_id
+        )
+        return resource.get('StackResourceDetail').get('PhysicalResourceId')
+
+    def get_stack_params(self, stack_name, param_name):
+        """
+        Get value of a particular param
+
+        Args:
+            stack_name (str): AWS cloudformation stack name
+            param_name (str): Stack parameter name
+
+        Returns:
+            str: Parameter value
+
+        """
+        stack_description = self.cf_client(
+            StackName=stack_name
+        )
+        params = stack_description.get('Stacks')[0].get('Parameters')
+        for param_dict in params:
+            if param_dict.get('ParameterKey') == param_name:
+                return param_dict.get('ParameterValue')
+
+    def get_worker_ignition_location(self, stack_name):
+        """
+        Get the ignition location from given stack
+
+        Args:
+            stack_name (str): AWS cloudformation stack name
+
+        Returns:
+            ignition_location (str): An AWS URL ignition location
+
+        """
+        param_name = 'IgnitionLocation'
+        ignition_loction = self.get_stack_params(
+            stack_name, param_name
+        )
+        return ignition_loction
+
+    def get_worker_instance_profile_name(self, stack_name):
+        """
+        Get the worker instance profile name
+
+        Args:
+            stack_name (str): AWS cloudformation stack name
+
+        Returns:
+            worker_instance_profile_name (str): instance profile name
+
+        """
+        param_name = 'WorkerInstanceProfileName'
+        worker_instance_profile_name = self.get_stack_params(
+            stack_name, param_name
+        )
+        return worker_instance_profile_name
+
+    def get_worker_stacks(self):
+        """
+        Get the cloudformation stacks only for workers of this cluster
+
+        Returns:
+            list : of worker stacks
+
+        """
+        worker_pattern = r"{}-no[0-9]*".format(config.ENV_DATA['cluster_name'])
```

Comment:
what if we need more than 10 workers?

Reply:
@ebenahar ,
This re matches 0 or more digits in front of 'no' so there is no limit , it matches any integer. However [0-9]* will match even just 'no', so I will change it to [0-9]+.

Here is an example:
```python
worker_pattern = r"{}-no[0-9]+".format('shmohanupi')
matching_stacks=[] 
all_stacks = ['shmohanupi-no1000', 'shmohanupi-no500','shmohanupi-no10','shmohanupi-no5','shmohanupi-no0','shmohanupi-no','coadyupi-no0','coadyupi-no1','shmohanupi-ma','elad-no0','elad-no1']

for stack in all_stacks:
    matching = re.match(worker_pattern, stack)
    if matching:
        matching_stacks.append(matching.group())

Matching stacks are 
-------------------------------------
['shmohanupi-no1000',
 'shmohanupi-no500',
 'shmohanupi-no10',
 'shmohanupi-no5',
 'shmohanupi-no0']
```